File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/repository/CredentialRepository.java
package com.example.chat.auth.server.core.repository;

import com.example.chat.auth.server.core.domain.Credential;
import com.example.chat.auth.server.core.domain.CredentialType;

import java.util.Optional;
import java.util.UUID;

/**
 * Credential 저장소
 */
public interface CredentialRepository {
    Optional<Credential> findByPrincipalId(UUID principalId, CredentialType credentialType);

    void save(UUID principalId, Credential credential);

    void delete(UUID principalId, CredentialType credentialType);

    boolean hasCredential(UUID principalId, CredentialType credentialType);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/repository/CredentialRepository.kt
package com.example.chat.auth.server.core.repository

import com.example.chat.auth.server.core.domain.Credential
import java.util.*

/** Credential 저장소 */
interface CredentialRepository {
    fun findByPrincipalId(principalId: UUID, credentialType: String): Optional<Credential>

    fun save(principalId: UUID, credential: Credential)

    fun delete(principalId: UUID, credentialType: String)

    fun hasCredential(principalId: UUID, credentialType: String): Boolean
}
